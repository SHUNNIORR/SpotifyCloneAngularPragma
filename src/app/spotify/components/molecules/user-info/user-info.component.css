.userinfo__container{
    padding: 10px 20px;
    height: auto;
    width: 100%;
    display: flex;
    background-color: rgb(17, 17, 17);
    justify-content: center;
    align-items: center;
    color: white;
    gap: 20px;
}
.userinfo__imgprofile{
    width: 50px;
    border-radius: 50%;
}
.userinfo__imgcontainer{
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
}
p{
    margin-bottom: 0;
}
.userinfo__email{
    color: rgb(146, 146, 146);
    font-family: SpotifyFontLight;
}