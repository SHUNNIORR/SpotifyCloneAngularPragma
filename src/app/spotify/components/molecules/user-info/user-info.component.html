
<div class="userinfo__container">
  <div *ngIf="thereInfo===false; else elseBlock" >
    <p userinfo__username>cargando...</p>
  </div>
    <ng-template #elseBlock>
    <div class="userinfo__imgcontainer">
      <a href="{{ infoUser.external_urls?.spotify! }}" target="_blank">
        <img
          src="{{infoUser!.images?.[0]?.url}}"
          alt=""
          class="userinfo__imgprofile"
        />
      </a>
    </div>
    <div>
      <p class="userinfo__username">{{ infoUser.display_name }}</p>
      <p class="userinfo__email">{{ infoUser.email }}</p>
    </div>
  </ng-template>
  </div>
