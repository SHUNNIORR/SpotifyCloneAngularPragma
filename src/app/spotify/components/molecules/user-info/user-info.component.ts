import { SpotifyApiService } from './../../../../services/spotify-api.service';
import { AuthService } from './../../../../services/auth.service';
import { Component, OnInit } from '@angular/core';
import { UserInfoInterface } from 'src/app/spotify/interfaces/userinfo.interface';


@Component({
  selector: 'app-user-info',
  templateUrl: './user-info.component.html',
  styleUrls: ['./user-info.component.css']
})
export class UserInfoComponent implements OnInit {

  constructor(private spotifyApi: SpotifyApiService, private authService: AuthService) { }
  thereInfo: boolean = false;
  infoUser={} as UserInfoInterface;
  ngOnInit(): void {
    console.log(this.infoUser);
    this.spotifyApi.getUserInfo().subscribe({
      next:(resp)=>{
        this.infoUser = resp
        this.thereInfo = true;
      },error:(error:any)=>{
        console.log(error);
        this.spotifyApi.alertCloseSession();
      }
   })
  }

}
