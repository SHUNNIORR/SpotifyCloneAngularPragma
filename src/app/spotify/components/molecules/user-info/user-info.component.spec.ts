import { AuthService } from './../../../../services/auth.service';
import { HttpClient, HttpHandler, HttpClientModule } from '@angular/common/http';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import {UserInfoComponent}from './user-info.component'


describe('User-info tests', () => {
    let component: UserInfoComponent;
    let fixture: ComponentFixture<UserInfoComponent>;
    let compiled:HTMLElement;
    beforeEach(() => {
        TestBed.configureTestingModule({
            declarations:[UserInfoComponent,],
            imports:[],
            providers: [ AuthService, HttpClient, HttpHandler]
        });
      });
      beforeEach(() => {
        fixture = TestBed.createComponent(UserInfoComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
        compiled = fixture.nativeElement
      })

    // test('should render my name', () => {
    //     fixture = TestBed.createComponent(UserInfoComponent);
    //     component = fixture.componentInstance;
    //     fixture.detectChanges();
    //     const compiled = fixture.nativeElement as HTMLInputElement;

    //     const pharagraph = compiled.querySelector('.userinfo__username')

    //     expect(pharagraph).toBeTruthy();
    // })
    test('Debe ser igual al snapshot', ()=>{
        expect(compiled).toMatchSnapshot();
    })
})