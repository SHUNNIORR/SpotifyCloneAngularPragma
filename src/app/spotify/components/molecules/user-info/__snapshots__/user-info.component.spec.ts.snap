// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User-info tests Debe ser igual al snapshot 1`] = `
<div
  id="root0"
>
  <div
    class="userinfo__container"
  >
    <div>
      <p
        userinfo__username=""
      >
        cargando...
      </p>
    </div>
    
    
  </div>
</div>
`;
